moduleframework/pdc_data.py

```diff
@@ -299,13 +300,24 @@ def createLocalRepoFromKoji(self):
                 (absdir, absdir), shell=True, verbose=is_debug())
         return "file://%s" % absdir
 
-
-
-odcs = ODCS("https://odcs.fedoraproject.org", auth_mech=AuthMech.OpenIDC, openidc_token="XXXXXXXXXXXXXXX")
-compose = odcs.new_compose("httpd-master", "module")
-compose = odcs.wait_for_compose(compose["id"], timeout=600)
-
-if compose["state_name"] == "done":
-    print "Compose done, URL with repo file", compose["result_repofile"]
-else:
-    print "Failed to generate compose"
+    def get_module_identifier(self):
+        if self.version:
+            return "%s-%s-%s" % (self.name, self.stream, self.version)
+        elif self.stream:
+            return "%s-%s" % (self.name, self.stream)
+        else:
+            return "%s-%s" % (self.name, "master")
+
+    def get_odcs_compose_repo(self, auth_token, compose_type="module"):
+        odcs = ODCS(ODCS_URL, auth_mech=AuthMech.OpenIDC, openidc_token=auth_token)
```

Comment:
Would probably deserve a dedicated class on its own.

Reply:
Sounds well, probably we need to rewrite this pdc_data.py completely, actually there are mixed koji, modulemd, pdc, odcs handling. 
It is caused by that complete module info is not in PDC, but composed by using more resources, pdc and koji and modulemds.
I can imagine that we'll have there some generic class for pdc, what will be extended by `koji`, second class extended by `odcs`. and then choose proper class based on `envvars`, does that sense @TomasTomecek

Reply:
yup, makes sense to separate them